<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Shared parameter documentation — params • acidroxygen</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Shared parameter documentation — params" />
<meta property="og:description" content="Shared parameter documentation" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">acidroxygen</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/acidgenomics/acidroxygen">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Shared parameter documentation</h1>
    <small class="dont-index">Source: <a href='https://github.com/acidgenomics/acidroxygen/blob/master/R/params.R'><code>R/params.R</code></a></small>
    <div class="hidden name"><code>params.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Shared parameter documentation</p>
    </div>


    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>Object.</p></td>
    </tr>
    <tr>
      <th>x</th>
      <td><p>Object.</p></td>
    </tr>
    <tr>
      <th>y</th>
      <td><p>Object.</p></td>
    </tr>
    <tr>
      <th>BPPARAM</th>
      <td><p><code>bpparamClass</code>.
BiocParallel parameter to specify the desired processor configuration.<br />
We recommend using one of the following:</p><ul>
<li><p><a href='https://rdrr.io/pkg/BiocParallel/man/register.html'>bpparam</a>.</p></li>
<li><p><a href='https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html'>SerialParam</a>.</p></li>
<li><p><a href='https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html'>MulticoreParam</a>.</p></li>
</ul></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code> or <code>NULL</code>.
Adjusted P value ("alpha") cutoff. If left <code>NULL</code>, will use the cutoff
defined in the object.</p></td>
    </tr>
    <tr>
      <th>asis</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Set this to <code>TRUE</code> when using the function inside a loop or inside an R
Markdown chunk with '<code>results = "asis"</code>' enabled.</p></td>
    </tr>
    <tr>
      <th>aspectRatio</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Aspect ratio.</p></td>
    </tr>
    <tr>
      <th>assay</th>
      <td><p><code><a href='https://rdrr.io/r/base/vector.html'>vector(1)</a></code>.
Assay name or index position.</p></td>
    </tr>
    <tr>
      <th>assays</th>
      <td><p><code>SimpleList</code>.
Count matrices, which must have matching dimensions. Counts can be passed
in as either a dense matrix (<code>matrix</code>) or sparse matrix (<code>sparseMatrix</code>).</p></td>
    </tr>
    <tr>
      <th>base</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Logarithm base.</p></td>
    </tr>
    <tr>
      <th>borderColor</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code> or <code>NULL</code>.
Border color.</p></td>
    </tr>
    <tr>
      <th>captions</th>
      <td><p><code>character</code> or <code>NULL</code>.
Captions.
If <code>NULL</code>, the names of the primary input list will be used automatically.</p></td>
    </tr>
    <tr>
      <th>cells</th>
      <td><p><code>character</code>.
Cell identifiers.</p></td>
    </tr>
    <tr>
      <th>censorSamples</th>
      <td><p><code>character</code>.
Specify a subset of samples to censor.</p></td>
    </tr>
    <tr>
      <th>class</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Object class.</p></td>
    </tr>
    <tr>
      <th>classes</th>
      <td><p><code>character</code>.
Object classes.</p></td>
    </tr>
    <tr>
      <th>closed</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(2)</a></code>.
Should the lower (1) and upper (2) bounaries be closed?</p></td>
    </tr>
    <tr>
      <th>colData</th>
      <td><p><code>DataFrame</code>.
Metadata describing the assay columns. For bulk RNA-seq, this data
describes the samples. For single-cell RNA-seq, this data describes the
cells.</p></td>
    </tr>
    <tr>
      <th>colnames</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Apply to column names.</p></td>
    </tr>
    <tr>
      <th>color</th>
      <td><p><code>ScaleDiscrete</code>.
Desired ggplot2 color scale. Must supply discrete values. When set <code>NULL</code>,
the default ggplot2 color palette will be used. If manual color definitions
are desired, we recommend using <code><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>ggplot2::scale_color_manual()</a></code>.</p>
<p>To set the discrete color palette globally, use:</p><pre>options(acid.color.discrete = ggplot2::scale_color_viridis_d())
</pre></td>
    </tr>
    <tr>
      <th>compress</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code> or <code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
These character strings are currently allowed for <code><a href='https://rdrr.io/r/base/save.html'>save()</a></code>:
<code>"gzip"</code>, <code>"bzip2"</code>, or <code>"xz"</code>.</p></td>
    </tr>
    <tr>
      <th>counts</th>
      <td><p><code>matrix</code>.
Count matrix. Normalized counts are recommended.</p></td>
    </tr>
    <tr>
      <th>countsAxisLabel</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Counts axis label.</p></td>
    </tr>
    <tr>
      <th>dark</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Plot against a dark background using the
<code><a href='https://rdrr.io/pkg/acidplots/man/themes.html'>acid_theme_light()</a></code> ggplot2 theme.</p></td>
    </tr>
    <tr>
      <th>dims</th>
      <td><p><code>integer</code>.
Vector of length 2 that denotes the columns from the reduced dimension
matrix to use for <code>centerX</code> and <code>centerY</code> column calculations. Defaults the
first and second dimensions.</p></td>
    </tr>
    <tr>
      <th>dir</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Directory path.</p></td>
    </tr>
    <tr>
      <th>direction</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Plot <code>"both"</code>, <code>"up"</code>, or <code>"down"</code> directions.</p></td>
    </tr>
    <tr>
      <th>ensemblRelease, release</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Ensembl release version (e.g. <code>90</code>). We recommend setting this value if
possible, for improved reproducibility. When left unset, the latest release
available via AnnotationHub/ensembldb is used. Note that the latest version
available can vary, depending on the versions of AnnotationHub and
ensembldb in use.</p></td>
    </tr>
    <tr>
      <th>envir</th>
      <td><p><code>environment</code>.
Environment.</p></td>
    </tr>
    <tr>
      <th>expression</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Calculation to apply.
Uses <code><a href='https://rdrr.io/r/base/match.arg.html'>match.arg()</a></code> internally and defaults to the first
argument in the <code>character</code> vector.</p></td>
    </tr>
    <tr>
      <th>FUN</th>
      <td><p><code>function</code>.</p></td>
    </tr>
    <tr>
      <th>file</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
File path.</p></td>
    </tr>
    <tr>
      <th>fill</th>
      <td><p><code>ggproto</code>/<code>ScaleDiscrete</code>.
Desired ggplot2 fill scale. Must supply discrete values. When set to
<code>NULL</code>, the default ggplot2 color palette will be used. If manual color
definitions are desired, we recommend using <code><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>ggplot2::scale_fill_manual()</a></code>.</p>
<p>To set the discrete fill palette globally, use:</p><pre>options(acid.fill.discrete = ggplot2::scale_fill_viridis_d())
</pre></td>
    </tr>
    <tr>
      <th>flip</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Flip x and y axes. Recommended for plots containing many samples.</p></td>
    </tr>
    <tr>
      <th>fun</th>
      <td><p><code>function</code>.</p></td>
    </tr>
    <tr>
      <th>gene2symbol</th>
      <td><p><code>Gene2Symbol</code>.
Gene-to-symbol mappings. Must contain <code>geneID</code> and <code>geneName</code> columns. See
<code>Gene2Symbol</code> for more information.</p></td>
    </tr>
    <tr>
      <th>genes</th>
      <td><p><code>character</code>.
Gene identifiers.</p></td>
    </tr>
    <tr>
      <th>genomeBuild</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Ensembl genome build assembly name (e.g. <code>"GRCh38"</code>). If set <code>NULL</code>,
defaults to the most recent build available.
Note: don't pass in UCSC build IDs (e.g. <code>"hg38"</code>).</p></td>
    </tr>
    <tr>
      <th>gffFile</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
GFF/GTF (General Feature Format) file.
Generally, we recommend using GTF (GFFv2) instead of GFFv3.</p></td>
    </tr>
    <tr>
      <th>geom</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Plot type. Uses <code><a href='https://rdrr.io/r/base/match.arg.html'>match.arg()</a></code> internally and defaults to
the first argument in the <code>character</code> vector.</p></td>
    </tr>
    <tr>
      <th>headerLevel</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code> (<code>1</code>-<code>7</code>).
Markdown header level.</p></td>
    </tr>
    <tr>
      <th>i</th>
      <td><p>Indices specifying elements to extract or replace. Indices are <code>numeric</code> or
<code>character</code> vectors, empty (<code>missing</code>), or <code>NULL</code>.</p>
<p>For more information:</p><pre>help(topic = "Extract", package = "base")
</pre></td>
    </tr>
    <tr>
      <th>ignoreTxVersion</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>. Don't the include the transcript version
in the identifier. Only applies when <code>level = "transcripts"</code>. This
simplifies identifier matching when generating a <code>tx2gene</code> file.</p></td>
    </tr>
    <tr>
      <th>inherits</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Should the enclosing frames of the <code>environment</code> be searched?</p></td>
    </tr>
    <tr>
      <th>interestingGroups</th>
      <td><p><code>character</code>.
Groups of interest to use for visualization.
Corresponds to factors describing the columns of the object.</p></td>
    </tr>
    <tr>
      <th>j</th>
      <td><p>Indices specifying elements to extract or replace. Indices are <code>numeric</code> or
<code>character</code> vectors, empty (<code>missing</code>), or <code>NULL</code>.</p>
<p>For more information:</p><pre>help(topic = "Extract", package = "base")
</pre></td>
    </tr>
    <tr>
      <th>label</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Superimpose sample text labels on the plot.</p></td>
    </tr>
    <tr>
      <th>labelSize</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Size of the text label.</p></td>
    </tr>
    <tr>
      <th>labels</th>
      <td><p><code>list</code>.
ggplot2 labels.
See <code><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggplot2::labs()</a></code> for details.</p></td>
    </tr>
    <tr>
      <th>lfcThreshold</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code> or <code>NULL</code>.
Log fold change ratio (base 2) cutoff threshold. If left <code>NULL</code>, will use
the cutoff defined in the object.</p></td>
    </tr>
    <tr>
      <th>level</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Return as genes or transcripts.</p></td>
    </tr>
    <tr>
      <th>legend</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Include plot legend.</p></td>
    </tr>
    <tr>
      <th>limit</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code>.
Threshold limit.</p></td>
    </tr>
    <tr>
      <th>lower</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code>.
Lower boundary.</p></td>
    </tr>
    <tr>
      <th>max</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code>.
Recommended maximum value cutoff.</p></td>
    </tr>
    <tr>
      <th>metadata</th>
      <td><p><code>list</code>.
Metadata.</p></td>
    </tr>
    <tr>
      <th>metric</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Metric type, either <code>length</code> or <code>elements</code>.</p></td>
    </tr>
    <tr>
      <th>min</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code>.
Recommended minimum value cutoff.</p></td>
    </tr>
    <tr>
      <th>minCounts</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Minimum number of counts per feature (i.e. gene).</p></td>
    </tr>
    <tr>
      <th>msg</th>
      <td><p><code>NULL</code> or <code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Custom message to return.</p></td>
    </tr>
    <tr>
      <th>n</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Number to include.</p></td>
    </tr>
    <tr>
      <th>ntop</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Number of top features to label.</p></td>
    </tr>
    <tr>
      <th>nullOK</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
If set to <code>TRUE</code>, <code>x</code> may also be <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>ordered</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.</p></td>
    </tr>
    <tr>
      <th>organism</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Full Latin organism name (e.g. <code>"Homo sapiens"</code>).</p></td>
    </tr>
    <tr>
      <th>overwrite</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Overwrite existing file on disk.</p></td>
    </tr>
    <tr>
      <th>pattern</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Pattern to use for matching.</p></td>
    </tr>
    <tr>
      <th>perMillion</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Display as counts per million.</p></td>
    </tr>
    <tr>
      <th>perSample</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Visualize the distributions per sample.</p></td>
    </tr>
    <tr>
      <th>pointAlpha</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code> (<code>0</code>-<code>1</code>).
Alpha transparency level.</p></td>
    </tr>
    <tr>
      <th>pointColor</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Default point color for the plot.</p></td>
    </tr>
    <tr>
      <th>pointScalar</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Default point size for the plot.</p></td>
    </tr>
    <tr>
      <th>pointSize</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code>.
Point size for dots in the plot.
In the range of 1-3 is generally recommended.</p></td>
    </tr>
    <tr>
      <th>pointsAsNumbers</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Plot the points as numbers (<code>TRUE</code>) or dots (<code>FALSE</code>).</p></td>
    </tr>
    <tr>
      <th>prefilter</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Apply prefiltering. Remove zero count genes.</p></td>
    </tr>
    <tr>
      <th>progress</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Show progress, using progress bars.</p></td>
    </tr>
    <tr>
      <th>projectDir</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Project directory path.</p></td>
    </tr>
    <tr>
      <th>reducedDims</th>
      <td><p><code>SimpleList</code>.
List containing matrices of cell coordinates in reduced space.</p></td>
    </tr>
    <tr>
      <th>reduction</th>
      <td><p><code><a href='https://rdrr.io/r/base/vector.html'>vector(1)</a></code>.
Dimension reduction name or index position.</p></td>
    </tr>
    <tr>
      <th>removeNA</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Remove <code>NA</code> values from calculations.</p></td>
    </tr>
    <tr>
      <th>return</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Return type. Uses <code><a href='https://rdrr.io/r/base/match.arg.html'>match.arg()</a></code> internally and defaults
to the first argument in the <code>character</code> vector.</p></td>
    </tr>
    <tr>
      <th>rowData</th>
      <td><p><code>DataFrame</code>.
Metadata describing the assay rows, if genomic ranges are not available.
<em>Use rowRanges (GRanges) instead, if possible</em>.</p></td>
    </tr>
    <tr>
      <th>rowRanges</th>
      <td><p><code>GRanges</code> or <code>GRangesList</code>.
Genomic ranges (e.g. genome annotations). Metadata describing the assay
rows.</p></td>
    </tr>
    <tr>
      <th>rownames</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Apply to row names.</p></td>
    </tr>
    <tr>
      <th>sampleMetadataFile</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Sample metadata file path. CSV or TSV is preferred, but Excel worksheets
are also supported. Check the documentation for conventions and required
columns.</p></td>
    </tr>
    <tr>
      <th>samples</th>
      <td><p><code>character</code>.
Sample identifiers.</p></td>
    </tr>
    <tr>
      <th>sigPointColor</th>
      <td><p><code>character</code>.
Color names for labeling upregulated and downregulated genes. Also supports
a character string for labeling DEGs with the same color, regardless of
direction.</p></td>
    </tr>
    <tr>
      <th>sort</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Resort using <code>sort</code>.</p></td>
    </tr>
    <tr>
      <th>spikeNames</th>
      <td><p><code>character</code>.
Vector indicating which assay rows denote spike-in sequences (e.g. ERCCs).</p></td>
    </tr>
    <tr>
      <th>subtitle</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Subtitle.</p></td>
    </tr>
    <tr>
      <th>tabset</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Include tabset marker.</p></td>
    </tr>
    <tr>
      <th>text</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Text.</p></td>
    </tr>
    <tr>
      <th>title</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Title.</p></td>
    </tr>
    <tr>
      <th>traceback</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Include traceback in error message.
See <code><a href='https://rdrr.io/r/base/traceback.html'>traceback()</a></code> for details.
<code><a href='https://rlang.r-lib.org/reference/entrace.html'>rlang::entrace()</a></code> also works nicely and can be set in <code>.Rprofile</code>.</p></td>
    </tr>
    <tr>
      <th>trans</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Name of the axis scale transformation to apply.</p>
<p>For more information:</p><pre>help(topic = "scale_x_continuous", package = "ggplot2")
</pre></td>
    </tr>
    <tr>
      <th>transgeneNames</th>
      <td><p><code>character</code>.
Vector indicating which assay rows denote transgenes (e.g. EGFP, TDTOMATO).</p></td>
    </tr>
    <tr>
      <th>trendline</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Include trendline on plot.</p></td>
    </tr>
    <tr>
      <th>tx2gene</th>
      <td><p><code>Tx2Gene</code>.
Transcript-to-gene mappings.</p></td>
    </tr>
    <tr>
      <th>unicode</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Allow Unicode characters in console output.</p></td>
    </tr>
    <tr>
      <th>uploadDir</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Final upload directory path.</p></td>
    </tr>
    <tr>
      <th>upper</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code>.
Upper boundary.</p></td>
    </tr>
    <tr>
      <th>url</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Uniform Resource Locator (URL).</p></td>
    </tr>
    <tr>
      <th>value</th>
      <td><p>Value to assign.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Run the function with verbose output.</p></td>
    </tr>
    <tr>
      <th>yaml</th>
      <td><p><code>list</code>.
YAML metadata.</p></td>
    </tr>
    <tr>
      <th>zeroPropagate</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Allow propagation of zeroes.</p></td>
    </tr>
    <tr>
      <th>.xname</th>
      <td><p>Name of object defined in <code>x</code>.
<em>Not intended to be used directly.</em></p></td>
    </tr>
    <tr>
      <th>.yname</th>
      <td><p>Name of object defined in <code>y</code>.
<em>Not intended to be used directly.</em></p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>No value.</p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://acidgenomics.com/'>Acid Genomics</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


